pages/[[...slug]]/index.tsx

```diff
@@ -0,0 +1,35 @@
+import { getAllPathnames, getPageContentByPathname } from '../../utilities/api'
+import type { GetStaticPaths, GetStaticProps } from 'next'
+import PageContent from '../../components/Main/PageContent'
+import type { PageContentProps } from '../../components/Main/PageContent'
+import React from 'react'
+import type { Pathname } from '../../utilities/api'
+
+type SitePageProps = {
+  pageContent: PageContentProps
+}
+
+export default function SitePage({ pageContent }: SitePageProps): JSX.Element {
+  return <PageContent content={pageContent.content} title={pageContent.title} />
+}
+
+export const getStaticProps: GetStaticProps = async ({ params }) => {
+  const pathname: string = (params?.slug as string[])?.join('/') ?? 'index'
+  const pageContent = await getPageContentByPathname(pathname)
+
+  return { props: { pageContent } }
+}
+
+export const getStaticPaths: GetStaticPaths = async () => {
+  type ParamWithSlug = { params: { slug: string[] } }
+
+  const allPathnames = (await getAllPathnames()) as Pathname[]
```

Comment:
I'm interested as to why we need to cast the return value of our own utility function here. Shouldn't we just be returning the type we expect from our own function?

Reply:
`getAllPathname()` is a API utility, and there could be a possibility on record in the Contentful, so it returns `Promise<Pathname[] | []>`.

Reply:
I learned from Chas's new code and realized `getAllPathname()` should returns `Promise<Pathname[]>` not `Promise<Pathname[] | []>`. This is not a valid discussion now. Thanks for the advice.